src/main/java/edu/wisc/my/personalizedredirection/controller/PersonalizedRedirectionController.java

```diff
@@ -79,7 +82,13 @@ public void setRedirectionService(IRedirectionService redirectionService) {
             logger.error("Issues happened while trying to generate custom link",
                     e);
             response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            hasError = true;
          }
+
+         if(hasError){
+            response.sendRedirect(errorURL);
```

Comment:
What response code will the response end up having? 404 because of the `SC_NOT_FOUND`? Or 302 because of the redirect?

Redirect on not found is confusing for its marring the location in the browser making it less clear what path wasn't found. Would it be feasible to get an attractive 404 experience to render at the originally requested not found path?